code/modules/power/lighting/light.dm

```diff
@@ -542,8 +543,10 @@
 
 /obj/machinery/light/proc/drop_light_tube(mob/user)
 	var/obj/item/light/light_object = new light_type()
+	if(reagents)
+		reagents.copy_to(light_object.reagents, LIGHT_REAGENT_CAPACITY)
+		QDEL_NULL(reagents)
```

Comment:
why not use ``trans_to()`` isntead?

Reply:
The QDEL_NULL (at least on this line) ensures that the tradeoff between the light tube and the machine doesn't leave anything behind. Removing it causes every light with a reagent lightbulb to look like this after it gets removed, swapped, etc. 

![sleight of hand](https://user-images.githubusercontent.com/28870487/190525795-b76122e3-3f74-4dad-b611-c4f3a2f06850.PNG)